controllers/menu_controller.rb

```diff
@@ -0,0 +1,149 @@
+# #1
+ require_relative "../models/address_book"
+ 
+ class MenuController
+   attr_accessor :address_book
+ 
+   def initialize
+     @address_book = AddressBook.new
+   end
+ 
+   def main_menu
+ # #2
+     puts "Main Menu - #{@address_book.entries.count} entries"
+     puts "1 - View all entries"
+     puts "2 - Create an entry"
+     puts "3 - Search for an entry"
+     puts "4 - Import entries from a CSV"
+     puts "5 - View Entry by entry number"
+     puts "6 - Exit"
+     print "Enter your selection: "
+    
+     
+ 
+ # #3
+     selection = gets.to_i
+     #puts "You picked #{selection}"
+   
+    # #7    
+     case selection
+     when 1
+       system "clear"
+       view_all_entries
+       main_menu
+     when 2
+       system "clear"
+       create_entry
+       main_menu
+     when 3
+       system "clear"
+       search_entries
+       main_menu
+     when 4
+       system "clear"
+       read_csv
+       main_menu
+     when 5
+       system "clear"
+       view_entry_number
+       main_menu
+     when 6
+       puts "Good-bye!"
+ # #8
+       exit(0)
+ # #9
+     else
+       system "clear"
+       puts "Sorry, that is not a valid input"
+       main_menu
+     end
+   end
+ 
+ # #10
+   def view_all_entries
+    @address_book.entries.each do |entry|
+      #system "clear"
+      puts entry.to_s
+    end  
+ # #15
+       #entry_submen6u(entry)
+     #end5
+ 
+     #system "clear"
+     puts "End of entries"
+   end
+ 
+   def create_entry
+     system "clear"
+     puts "New AddressBloc Entry"
+ # #12
+     print "Name: "
+     name = gets.chomp
+     print "Phone number: "
+     phone = gets.chomp
+     print "Email: "
+     email = gets.chomp
+ 
+ # #13
+     @address_book.add_entry(name, phone, email)
+ 
+     system "clear"
+     puts "New entry created"
+   end
+ 
+   def search_entries
+    
+   end
+ 
+   def read_csv
+   end
+   def view_entry_number
+ 
+     print "Input the entry number of the AddressBook entry you wish to see: "
+     
+     entry_number = gets.to_i
+     a = *(1..@address_book.entries.count)
+     puts '#{a}'
```

Comment:
What output this to the screen?  Also I recommend using meaningful variable names instead of generic ones like `a`.  People often say ruby code should be "self-documenting" meaning you can read the code and quckly understand what it is doing with no need for comments.